plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerLaunchConfigurationDelegate.java

```diff
@@ -299,6 +302,25 @@ DefaultRunConfiguration generateServerRunConfiguration(ILaunchConfiguration conf
       devServerRunConfiguration.setAdditionalArguments(programArguments);
     }
 
+    boolean environmentAppend =
+        configuration.getAttribute(ILaunchManager.ATTR_APPEND_ENVIRONMENT_VARIABLES, true);
+    if (!environmentAppend) {
+      throw new CoreException(StatusUtil.error(this, "'Replace environment' not yet supported"));
+    }
+    // Could use getEnvironment(), but it does `append` processing and joins as key-value pairs
+    Map<String, String> environment = configuration.getAttribute(
+        ILaunchManager.ATTR_ENVIRONMENT_VARIABLES, Collections.<String, String>emptyMap());
+    if (environment != null) {
```

Comment:
Probably we don't need this null check, as we return the empty set when it's undefined.